pkgs/development/python-modules/google_cloud_bigquery/default.nix

```diff
@@ -1,30 +1,29 @@
-{ stdenv
-, buildPythonPackage
-, fetchPypi
-, freezegun
-, google_resumable_media
-, google_api_core
-, google_cloud_core
-, pandas
-, pyarrow
-, pytest
-, mock
-, ipython
-}:
+{ stdenv, buildPythonPackage, fetchPypi, pythonOlder, freezegun, google_api_core
+, google_cloud_core, google_cloud_testutils, google_resumable_media, grpcio
+, ipython, mock, pandas, proto-plus, pyarrow, pytest }:
 
 buildPythonPackage rec {
   pname = "google-cloud-bigquery";
-  version = "1.26.1";
+  version = "2.1.0";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "51c29b95d460486d9e0210f63e8193691cd08480b69775270e84dd3db87c1bf2";
+    sha256 = "0x5g6n151rcdgq4s80f71zpsl7bsvyyrs07l58psdpyd3kwf4sbk";
   };
 
-  checkInputs = [ pytest mock ipython freezegun ];
-  propagatedBuildInputs = [ google_resumable_media google_api_core google_cloud_core pandas pyarrow ];
+  disabled = pythonOlder "3.6";
 
-  # prevent local directory from shadowing google imports
+  checkInputs = [ freezegun google_cloud_testutils ipython mock pytest ];
+  propagatedBuildInputs = [
+    google_resumable_media
+    google_api_core
+    google_cloud_core
+    pandas
+    proto-plus
+    pyarrow
+  ];
+
+  # prevent google directory from shadowing google imports
   # call_api_applying_custom_retry_on_timeout requires credentials
   # test_magics requires modifying sys.path
   checkPhase = ''
```

Comment:
maybe disable this test?

```
builder for '/nix/store/2nm6nj24sswksj2p5gw7myyddpbywg4l-python3.8-google-cloud-bigquery-2.1.0.drv' failed with exit code 1; last 10 log lines:
  E       datetime.datetime(4567, 1, 1, 0, 0)
  E
  E       + [datetime.datetime(4567, 1, 1, 0, 0), datetime.datetime(9999, 12, 31, 0, 0)]
  E       - [datetime.datetime(4567, 1, 1, 0, 0, tzinfo=<UTC>),
  E       -  datetime.datetime(9999, 12, 31, 0, 0, tzinfo=<UTC>)]

  tests/unit/test_table.py:2294: AssertionError
  =========================== short test summary info ============================
  FAILED tests/unit/test_table.py::TestRowIterator::test_to_dataframe_timestamp_out_of_pyarrow_bounds
  ========== 1 failed, 1372 passed, 54 skipped, 1 deselected in 14.60s ===========
```

Reply:
`pytestCheckHook` would make this easy

Reply:
Strange, all tests pass on my machine. Just checked again with `nix-build -A python38Packages.google_cloud_bigquery`. I can disable the test, though.

Reply:
Disabled.